### PR TITLE
Njb/use cross env

### PR DESCRIPTION
**Pull request by NickBolles**
### Summary
- [ ] Tell us about the problem your pull request is solving.
- [ ] Are there any open issues that are related to this?
- [ ] Is this PR dependent on PRs in other repos?

If so, please mention them to keep the conversations linked together.

### Other Information
Use cross-env to set environment variables so that the scripts don't bomb out on windows

### CODE REVIEW

**Comment by eddyystop** [comment]
Feathers uses shx ( https://github.com/shelljs/shx ) and I do.n't want to introduce another player.

Would you mind converting to shx?

**Comment by NickBolles** [comment]
@eddyystop sounds good. Looking into it briefly I'm not sure how I would do this with shx though. 

I found [this issue](https://github.com/shelljs/shx/issues/127) which references cross-env and it sounds like it's not really possible in shx.

Do you have any ideas?

**Comment by eddyystop** [comment]
@daffl ^^^

**Comment by eddyystop** [comment]
Is there any way at all to continue using shx?

Otherwise
- Do we address this issue?
- Do we introduce 2 overlapping utilities, shx and cross-env?
- Do we replace shx with cross-env? Since package.json scripts are not regenerated, old apps would be using shx, while new ones cross-env.

**Comment by j2L4e** [comment]
I have little experience with shx, but from what's said in https://github.com/shelljs/shx/issues/127 it seems like it can't replace `cross-env` functionality here

**Comment by eddyystop** [comment]
@NickBolles Could you confirm that you can't do what you want to with shx. Shoujld that be the case, could you resolve the merge conflicts and I will merge.

File: generators/writing/templates/src/seed-data.ejs [hunk 1]
```diff
@@ -10,8 +10,8 @@
 let ifSeedServices = ['--seed', '-s'].some(str => process.argv.slice(2).includes(str))<%- sc %>
 
 // Determine if environment allows test to mutate existing DB data.
-let env = (config.tests || {}).environmentsAllowingSeedData || []<%- sc %>
-let ifDbChangesAllowed = env.includes(process.env.NODE_ENV)<%- sc %>
+let { tests: { environmentsAllowingSeedData = [] } = {}} = config<%- sc %>
+let ifDbChangesAllowed = environmentsAllowingSeedData.includes(process.env.NODE_ENV)<%- sc %>
```
[the remark this answers is not in the document]
Comment (NickBolles) [reply generators/writing/templates/src/seed-data.ejs]: I changed this to be a little utility function that takes the config. This way we can pass the config from the app instance

**Comment by NickBolles** [comment]
@j2L4e I updated those changes. Thanks for looking at them!

File: generators/writing/templates/src/seed-data.ejs (hunk 1 above)
[the remark this answers is not in the document]
Comment (eddyystop) [reply generators/writing/templates/src/seed-data.ejs]: The thing I liked about `ifDbChangesAllowed ` was that the dev can refer to it in the custom code. Its simpler than having to call the function with app.get('tests'). That var name can be a standard throughout DB mutating tests.

I don't want to cause a fuss but the generated code should always be as flexible as possible for customizations.

[the remark this answers is not in the document]
Comment (NickBolles) [reply generators/writing/templates/src/seed-data.ejs]: I can undo this. But I don't really see how it makes things less flexible and it seems like importing the json directly will break config and introduce bugs. 

I definitely get what you are getting at, but this seems like a case where there's benefit to making this change, and that there's really one use to the variable, short circuit the seed data function

[the remark this answers is not in the document]
Comment (eddyystop) [reply generators/writing/templates/src/seed-data.ejs]: I agree the import should be replaced by app.get.

I was probably not clear enough. I like having ifDbChangesAllowed as an already set value for the user as it easier to find than the function.

It's possible the user remove the generated exit for something else and having a set value looks advantageous.

[the remark this answers is not in the document]
Comment (NickBolles) [reply generators/writing/templates/src/seed-data.ejs]: @eddyystop ahhh I see what you mean. Check my most recent commit for an update on this. Let me know if you were thinking anything different.

[the remark this answers is not in the document]
Comment (eddyystop) [reply generators/writing/templates/src/seed-data.ejs]: +1

**Comment by eddyystop** [comment]
Completed this PR with https://github.com/feathers-plus/generator-feathers-plus/pull/137

**Comment by NickBolles** [comment]
Thank for the help you guys!

**Pull request merged**